R/Ameriflux_data_tables.R

```diff
@@ -0,0 +1,562 @@
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data5 <- read.csv('/home/reevesman/Ameriflux/AMF_US-EML/AMF_US-EML_BASE_HH_1-4.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data6 <- read.csv('/home/reevesman/Ameriflux/AMF_US-Brw/AMF_US-Brw_BASE_HH_2-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data7 <- read.csv('/home/reevesman/Ameriflux/AMF_US-Atq/AMF_US-Atq_BASE_HH_1-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data8 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Qfo/AMF_CA-Qfo_BASE_HH_1-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data9 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Ojp/AMF_CA-Ojp_BASE_HH_1-1.csv',
+                  skip = 2,
+                  stringsAsFactors = F)
+
+data10 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Oas/AMF_CA-Oas_BASE_HH_1-1.csv',
+                   skip = 2,
+                   stringsAsFactors = F)
+
+data11 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Man/AMF_CA-Man_BASE_HH_2-1.csv',
+                   skip = 2,
+                   stringsAsFactors = F)
+
+data12 <- read.csv('/home/reevesman/Ameriflux/AMF_CA-Gro/AMF_CA-Gro_BASE_HH_1-1.csv',
+                   skip = 2,
+                   stringsAsFactors = F)
+
+
+##############################################################################
+
+#function definition for attribute_definitions
+  #inputs:
+    #data: a dataframe with attributes to define
+    #definitions: dataframe in the style of definitions.csv
+  #output: a vector of attribute definitions to be used in attribute table
+
+##############################################################################
+
+attribute_definitions <- function(data, definitions){
+  
+  attributes <- colnames(data)
+  
+  #will be vector of attribute definitions
+  defs <- vector(mode = 'character', length = length(attributes))
+  
+  #i will be index at which to populate defs
+  i <- 1
+  #for each attribute
+  for (att in attributes){
+    
+    #if the attribute has one of the attached qualifiers
+    #get definition of qualifier to paste onto regular definition later
+    #set att to the regular attribute name (w/out qualifier)
+    #set a flag that this attribute includes a qualifier
+    
+    #did not do all of the possible combinations of qualifiers
+    #because not all possible combinations appeared in data
+    
+    if (str_sub(att,-5,-1) == '_PI_F'){
+      x <- str_sub(att,-5,-1)
+      extra <- paste(definitions[which(definitions$uniqueAttributeLabel == '_PI'), 'uniqueAttributeDefinition'],
+                     definitions[which(definitions$uniqueAttributeLabel == '_F'), 'uniqueAttributeDefinition'],
+                     sep = '; ')
+      str_sub(att,-5,-1) <- ""
+      QUALIFIERS_EXIST <- TRUE
+    }
+    
+    else if (str_sub(att,-3,-1) %in% c('_PI','_QC','_IU','_SD')){
```

Comment:
It looks like you're looking for every possible iteration of qualifiers and treating each one as a unique case.  While this is totally acceptable, it's a good idea to think about how you might scale this if there were too many to write out by hand.  

You can reverse the %in% statement so you're looking for if a qualifier is in an attribute.  Then you can run the rest of the commands that you have after an else-if statement to get the `extra` definition.  

See if you can use this example to simplify your code.  You should only need to do this twice, once for the integer qualifiers and once for the character ones.  The logic here is a bit tricky so don't hesitate to ask me about it 

```
att <- "CO2_F_1
int_qualifiers <- c('_1','_2','_3','_4','_5','_6','_7','_8','_9')
sapply(int_qualifiers, grepl, x = att)   # this applies grepl to each value in int_qualifiers, with the additional argument x = att
```